src/main/java/net/wimpi/modbus/net/SerialConnection.java

```diff
@@ -338,8 +330,7 @@ public boolean isConnected() {
 
     @Override
     public String toString() {
-        return new ToStringBuilder(this, toStringStyle).append("portName", m_Parameters.getPortName())
-                .append("port", m_SerialPort).toString();
+        return "SerialConnection [m_SerialPort=" + m_SerialPort + ", m_Parameters.getPortName()="
```

Comment:
Risk of NPE

Reply:
Ditto. I used the Eclipse macro to create these methods, but it seems not to be aware of null checks. Odd.

Reply:
See above..

Reply:
However, here there is indeed a potential for NPE

Reply:
Fixed